‘SOUND_PCM_WRITE_CHANNELS_is_obsolete_use_SNDCTL_DSP_CHANNELS_instead’

### FluidSynth version
2.2.9 release

### Describe the bug
Can't build. During build on a Devuan GNU/Linux daedalus machine, I get:
```
[ 14%] Building C object src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_oss.c.o
In file included from /usr/include/x86_64-linux-gnu/sys/soundcard.h:1,
                 from /usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c:36:
/usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c: In function ‘new_fluid_oss_audio_driver’:
/usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c:233:26: error: ‘SOUND_PCM_WRITE_CHANNELS_is_obsolete_use_SNDCTL_DSP_CHANNELS_instead’ undeclared (first use in this function)
  233 |     if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c:233:26: note: each undeclared identifier is reported only once for each function it appears in
/usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c: In function ‘new_fluid_oss_audio_driver2’:
/usr/local/src/fluidsynth-2.2.9/src/drivers/fluid_oss.c:379:26: error: ‘SOUND_PCM_WRITE_CHANNELS_is_obsolete_use_SNDCTL_DSP_CHANNELS_instead’ undeclared (first use in this function)
  379 |     if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/libfluidsynth-OBJ.dir/build.make:132: src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_oss.c.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:241: src/CMakeFiles/libfluidsynth-OBJ.dir/all] Error 2
gmake: *** [Makefile:166: all] Error 2
```
Will provide additional info on request.

### Expected behavior
Build should conclude successfully

### Additional context
I wanted to use the Debian version of the package, but that seems to depend on systemd for some strange reason :-(

Still getting the same error:
```
In file included from /usr/include/x86_64-linux-gnu/sys/soundcard.h:1,
                 from /usr/local/src/fluidsynth/src/drivers/fluid_oss.c:36:
/usr/local/src/fluidsynth/src/drivers/fluid_oss.c: In function ‘new_fluid_oss_audio_driver’:
/usr/local/src/fluidsynth/src/drivers/fluid_oss.c:233:26: error: ‘SOUND_PCM_WRITE_CHANNELS_is_obsolete_use_SNDCTL_DSP_CHANNELS_instead’ undeclared (first use in this function)
  233 |     if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/src/fluidsynth/src/drivers/fluid_oss.c:233:26: note: each undeclared identifier is reported only once for each function it appears in
/usr/local/src/fluidsynth/src/drivers/fluid_oss.c: In function ‘new_fluid_oss_audio_driver2’:
/usr/local/src/fluidsynth/src/drivers/fluid_oss.c:379:26: error: ‘SOUND_PCM_WRITE_CHANNELS_is_obsolete_use_SNDCTL_DSP_CHANNELS_instead’ undeclared (first use in this function)
  379 |     if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/libfluidsynth-OBJ.dir/build.make:146: src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_oss.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:214: src/CMakeFiles/libfluidsynth-OBJ.dir/all] Error 2

--- a/src/drivers/fluid_oss.c
+++ b/src/drivers/fluid_oss.c
@@ -52,7 +52,7 @@
 #define BUFFER_LENGTH 512
 
 // Build issue on some systems (OSS 4.0)?
-#if !defined(SOUND_PCM_WRITE_CHANNELS) && defined(SNDCTL_DSP_CHANNELS)
+#if (defined(SOUND_VERSION) && SOUND_VERSION >= 0x040000) || (!defined(SOUND_PCM_WRITE_CHANNELS) && defined(SNDCTL_DSP_CHANNELS))
 #define SOUND_PCM_WRITE_CHANNELS        SNDCTL_DSP_CHANNELS
 #endif
 

--- a/src/drivers/fluid_oss.c
+++ b/src/drivers/fluid_oss.c
@@ -53,7 +53,9 @@
 
 // Build issue on some systems (OSS 4.0)?
 #if (defined(SOUND_VERSION) && SOUND_VERSION >= 0x040000) || (!defined(SOUND_PCM_WRITE_CHANNELS) && defined(SNDCTL_DSP_CHANNELS))
-#define SOUND_PCM_WRITE_CHANNELS        SNDCTL_DSP_CHANNELS
+#define OSS_CHANNELS_PLACEHOLDER        SNDCTL_DSP_CHANNELS
+#else
+#define OSS_CHANNELS_PLACEHOLDER        SOUND_PCM_WRITE_CHANNELS
 #endif
 
 /** fluid_oss_audio_driver_t
@@ -230,7 +232,7 @@ new_fluid_oss_audio_driver(fluid_settings_t *settings, fluid_synth_t *synth)
 
     channels = 2;
 
-    if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
+    if(ioctl(dev->dspfd, OSS_CHANNELS_PLACEHOLDER, &channels) < 0)
     {
         FLUID_LOG(FLUID_ERR, "Can't set the number of channels");
         goto error_recovery;
@@ -376,7 +378,7 @@ new_fluid_oss_audio_driver2(fluid_settings_t *settings, fluid_audio_func_t func,
 
     channels = 2;
 
-    if(ioctl(dev->dspfd, SOUND_PCM_WRITE_CHANNELS, &channels) < 0)
+    if(ioctl(dev->dspfd, OSS_CHANNELS_PLACEHOLDER, &channels) < 0)
     {
         FLUID_LOG(FLUID_ERR, "Can't set the number of channels");
         goto error_recovery;

